### ISSUE TITLE
EXECUTE IMMEDIATE FROM Parsing Error (SNOWFLAKE DIALECT)

### ISSUE THREAD

**Issue by markusos** [sqlfluff/sqlfluff#5694]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

We have a project where we use the `EXECUTE IMMEDIATE FROM` command (https://docs.snowflake.com/en/sql-reference/sql/execute-immediate-from) to execute staged sql files. 

This command executes fine in Snowflake without any issues, but fails to parse in sqlfuff.

This command is similar to the supported `EXECUTE IMMEDIATE` command that does parse correctly in sqlfluff. The fix is probably to extend the work done in https://github.com/sqlfluff/sqlfluff/pull/5275

### Expected Behaviour

`EXECUTE IMMEDIATE FROM` commands should not throw a parse error.

### Observed Behaviour

Running sqlfluff on a file with this content:
```
EXECUTE IMMEDIATE FROM './temp/create_db.sql';
```

Results in:
```
> $ sqlfluff lint --dialect snowflake
== [/snowflake/run.sql] FAIL
L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: "EXECUTE
                       | IMMEDIATE FROM './temp/create_db..."
WARNING: Parsing errors found and dialect is set to 'snowflake'. Have you configured your dialect correctly?
All Finished 📜 🎉!
```

The parsing output is as following:
```
> $ sqlfluff parse --dialect snowflake /snowflake/run.sql
[L:  1, P:  1]      |file:
[L:  1, P:  1]      |    unparsable:                                               !! Expected: "<Delimited: [<Ref: 'StatementSegment'>]>"
[L:  1, P:  1]      |        word:                                                 'EXECUTE'
[L:  1, P:  8]      |        whitespace:                                           ' '
[L:  1, P:  9]      |        word:                                                 'IMMEDIATE'
[L:  1, P: 18]      |        whitespace:                                           ' '
[L:  1, P: 19]      |        word:                                                 'FROM'
[L:  1, P: 23]      |        whitespace:                                           ' '
[L:  1, P: 24]      |        single_quote:                                         "'./temp/create_db.sql'"
[L:  1, P: 46]      |        semicolon:                                            ';'
[L:  1, P: 47]      |    newline:                                                  '\n'
[L:  2, P:  1]      |    [META] end_of_file:

==== parsing violations ====
L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: "EXECUTE IMMEDIATE
                       | FROM './temp/create_db..."
WARNING: Parsing errors found and dialect is set to 'snowflake'. Have you configured your dialect correctly?
```

### How to reproduce

Create a file with: 

```
EXECUTE IMMEDIATE FROM './temp/create_db.sql';
```

Run: 

```
sqlfluff lint --dialect snowflake
```

### Dialect

Snowflake

### Version

> $ sqlfluff --version
sqlfluff, version 3.0.2

> $ python --version
Python 3.11.4

### Configuration

None

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
